module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,51 @@
+#include "compression.hpp" 
+
+std::vector<std::pair<uint8_t, uint8_t>> compressOneLine(const  std::array<uint8_t,width>& line){
+
+    uint8_t value;
+    uint8_t previous_value=line[0];
```

Comment:
You can use front instead [0]

Reply:
Is it faster ?

Reply:
no just fornt() <--> *begin().

Reply:
OK, but what are the disadvantages of my method?

Reply:
For vector it's ok. However, to write more generic code, that requires less work in future is better to use `front()`. When someone will need to change a container for example to list this piece of code will require refactoring. With `front()` it will still work.